src/Reflection/ReflectionClass.php

```diff
@@ -1238,4 +1244,17 @@ public function removeProperty($propertyName)
 
         return false;
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
```

Comment:
There are multiple places where the context is envoked for different `TypeFinder\Find*` classes (`FindTypeFromAst`, `FindReturnType`, `FindParameterType`). An a `TypeFinder` class might look like:

```php
class TypeFinder
{
    public function findTypeForAst($ast);
    public function findTypeForReflectionParameter();
    public function findTypeForFunction();
    public function findType...
}
```

Would that be acceptable?